sqlglot/parser.py

```diff
@@ -1512,6 +1533,22 @@ def _parse_index(self) -> exp.Expression:
             columns=self._parse_expression(),
         )
 
+    def _parse_create_table_index(self):
+        unique = self._prev.token_type == TokenType.UNIQUE
+        primary = self._prev.text.upper() == "PRIMARY" or self._match_text_seq("PRIMARY")
```

Comment:
why do you check _prev.text == primary and match_text primary?

Reply:
The text could be either "PRIMARY INDEX" or "UNIQUE PRIMARY INDEX", and the first word has already been consumed when we hit this line.

If "primary" is the first word in the text it will be _prev and "index" will be _curr. If primary is the second word in the text, "unique" will be _prev and "primary" will be _curr.

Reply:
then you can do

self._match_text_seq("PRIMARY", "INDEX") 
self._match_text_seq("UNIQUE", "PRIMARY", "INDEX")

Reply:
Sorry, I worded that comment poorly - there are actually 6 distinct text sequences